internal/resolution/variablesources/bundle_uniqueness_test.go

```diff
@@ -92,23 +92,19 @@ func TestMakeBundleUniquenessVariables(t *testing.T) {
 	// of the involved bundles to be allowed in the solution
 	expectedVariables := []*olmvariables.BundleUniquenessVariable{
 		{
-			SimpleVariable: input.NewSimpleVariable(
-				"test-package package uniqueness",
-				constraint.AtMost(
-					1,
-					deppy.Identifier("fake-catalog-test-package-test-package.v1.0.0"),
-					deppy.Identifier("fake-catalog-test-package-test-package.v1.0.1"),
-				),
-			),
+			ID: "test-package package uniqueness",
```

Comment:
This is not the best example, but I'm trying to find a nicier way to do these comparisons.

E.g. in #498 we do this:

```go
	gocmpopts := []cmp.Option{
		cmpopts.IgnoreUnexported(sync.RWMutex{}),
		cmp.AllowUnexported(
			sync.RWMutex{},
			olmvariables.BundleVariable{},
			input.SimpleVariable{},
			constraint.DependencyConstraint{},
			catalogmetadata.Bundle{},
		),
		cmpopts.IgnoreFields(
			// Do not compare a func field
			catalogmetadata.PackageRequired{}, "SemverRange",
		),
	}
	require.Empty(t, cmp.Diff(bundles, expectedVariables, gocmpopts...))
```

Reply:
Haven't had enough time to review all of it - but you likely do *not* want to compare internals of `sync.RWMutex` - perhaps even of the others. For each, ask if it's possible or valid for the third-party package to change the implementation detail of unexported fields and if that would be a bona-fide concern for the caller here.